roles/ceph-infra/tasks/configure_firewall.yml

```diff
@@ -152,18 +152,35 @@
   tags:
     - firewall
 
-- name: open iscsi ports
+- name: open iscsi target ports
   firewalld:
-    port: "5001/tcp"
+    port: "3260/tcp"
     zone: "{{ ceph_iscsi_firewall_zone }}"
     source: "{{ public_network }}"
     permanent: true
     immediate: true
     state: enabled
   notify: restart firewalld
   when:
-    - iscsi_group_name is defined
-    - iscsi_group_name in group_names
+    - iscsi_gw_group_name is defined
+    - iscsi_gw_group_name in group_names
+    - (firewalld_pkg_query.get('rc', 1) == 0 or is_atomic)
+  tags:
+    - firewall
+
+- name: open iscsi api ports
+  firewalld:
+    port: "{{ api_port }}/tcp"
+    zone: "{{ ceph_iscsi_firewall_zone }}"
+    source: "{{ public_network }}"
+    permanent: true
+    immediate: true
+    state: enabled
+  notify: restart firewalld
+  when:
+    - api_port is defined
```

Comment:
`api_port` is always defined so we don't need this check.

Reply:
Are the defaults initialized before the firewall above?

If I do not have that check I get the error:


TASK [ceph-infra : open iscsi api ports] ***************************************
Tuesday 06 November 2018  21:37:15 -0600 (0:00:00.924)       0:00:18.323 ****** 
fatal: [ans6]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'api_port' is undefined\n\nThe error appears to have been in '/root/src/ceph-ansible/roles/ceph-infra/tasks/configure_firewall.yml': line 171, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: open iscsi api ports\n  ^ here\n"}

Reply:
Ah shoot, have we talked about this before? Do I need to do

port: "{{ api_port | default(5000) }}/tcp"

Reply:
My bad `api_port` actually comes from the ceph-iscsi role, I think the `port: "{{ api_port | default(5000) }}/tcp"` is what we want then.

Reply:
Fix up and repushed.